src/Storages/StorageBuffer.cpp

```diff
@@ -1010,14 +1012,12 @@ void StorageBuffer::checkAlterIsPossible(const AlterCommands & commands, Context
 std::optional<UInt64> StorageBuffer::totalRows(const Settings & settings) const
 {
     std::optional<UInt64> underlying_rows;
-    auto underlying = DatabaseCatalog::instance().tryGetTable(destination_id, getContext());
+    auto destination = DatabaseCatalog::instance().tryGetTable(destination_id, getContext());
 
-    if (underlying)
-        underlying_rows = underlying->totalRows(settings);
-    if (!underlying_rows)
-        return underlying_rows;
```

Comment:
Here was another bug, CC @azat.

Reply:
Indeed, nice catch. (Thread can be resolved).